public/app/plugins/datasource/loki/components/LokiQueryField.tsx

```diff
@@ -47,7 +47,7 @@ export class LokiQueryField extends React.PureComponent<LokiQueryFieldProps, Lok
     const refreshLabels = shouldRefreshLabels(range, prevProps.range);
     // We want to refresh labels when range changes (we round up intervals to a minute)
     if (refreshLabels) {
-      languageProvider.fetchLabels();
+      languageProvider.fetchLabels({ timeRange: range });
```

Comment:
I know you wanted to follow the `options` pattern, but I find it more _aesthetic_ and verbose to do:

```suggestion
      languageProvider.fetchLabels(range);
```

Reply:
I would rather keep `options pattern` as in the future, there might be other arguments that can be added. There is a for example discussions about fuzzy search and one of the proposal is to add it to `/labels` endpoint. So for clean and extendable APIs, I think it make sense to follow this pattern.

Reply:
Sounds good to me 👍